lib/util/source-code-fixer.js

```diff
@@ -68,9 +73,13 @@ SourceCodeFixer.applyFixes = function(sourceCode, messages) {
     // clone the array
     const remainingMessages = [],
         fixes = [],
+        bom = (sourceCode.hasBOM ? BOM : ""),
         text = sourceCode.text;
-    let lastFixPos = text.length + 1,
-        prefix = (sourceCode.hasBOM ? BOM : "");
+    let fix = null,
+        lastPos = 0,
+        start = 0,
+        end = 0,
```

Comment:
Can `start` and `end` be defined in a smaller scope? (They don't seem to be used outside of the `for-of` loop.)